lib/modules/landing/landing_interactor.dart

```diff
@@ -1,8 +1,24 @@
 part of 'landing_module.dart';
 
 abstract class LandingInteractor extends Interactor<LandingInteractorDelegate> {
+  void validateAuthentication();
 }
 
-abstract class LandingInteractorDelegate {}
+abstract class LandingInteractorDelegate {
+  BehaviorSubject<bool> get authenticationDidValidate;
 
-class LandingInteractorImpl extends LandingInteractor {}
+  BehaviorSubject<Object> get authenticationDidFailToValidate;
```

Comment:
I have an idea: how about grouping these 2 Subjects into 1, forming a new data class like this:

`BehaviorSubject<ValidateResult> get validateResult;`

so inside, ValidateResult actually has:
`Success extends ValidateResult` and `Failed extends ValidateResult`

doing this you will need only 1 stream; 
avoid the vague `<Object>` type, because you can specific `Error` as a property inside `Failed`.

How do you think?

Reply:
Yea, I gonna do it but have a little difference 😁 
```dart
BehaviorSubject<bool> get validateResult;

validateResult.listen(() {
  // handle success
}, onError: () {
  // handle exception
})
```

Then all unhandled error and exception will be handled in `main.dart`
```dart
return runZonedGuarded(() {
    runApp(App());
  }, (error, stack) {
    print("error: $error\nstack: $stack");
  });
```

Btw I will update this `BehaviorSubject<Object> get authenticationDidFailToValidate;` to `BehaviorSubject<Exception> get authenticationDidFailToValidate;`

Reply:
@markgravity I think we should avoid using `Object` class as much as possible, defining a specific type then we don't need to cast `Object` to a class later. 😂

Reply:
You're right, this `Object` should be `Exception`.
Let me update! 👍